src/qgis_geonode/gui/geonode_source_select_provider.py

```diff
@@ -54,51 +54,67 @@ def __init__(self, parent, fl, widgetMode):
         self.btnDelete.clicked.connect(self.delete_connection)
         self.toggle_connection_management_buttons()
 
-        settings_manager.current_connection_changed.connect(
+        connections_manager.current_connection_changed.connect(
             self.update_connections_combobox
         )
 
+        current_connection = connections_manager.get_current_connection()
+        if current_connection is None:
+            existing_connections = connections_manager.list_connections()
+            if len(existing_connections) > 0:
+                current_connection = existing_connections[0]
```

Comment:
Since they are being sorted by name (check the implementation of `conf.ConnectionManager.list_connections` this is a fairly arbitrary choice. Picking the first one or the last does not imply it is more recent.

Or are you suggesting we add a new temporal field to each `ConnectionSettings` instance to track when it was created and then sort by that? IMO it sounds interesting, but I don't think it is relevant to implement such a feature at the moment

Reply:
> Or are you suggesting we add a new temporal field to each `ConnectionSettings` instance to track when it was created and then sort by that?

no, I expected the connection list have the recent connection at the last index.

> IMO it sounds interesting, but I don't think it is relevant to implement such a feature at the moment

sure